docs/BreakingChanges/093 Linq to EF Join throws ArgumentOutOfRangeException.md

```diff
@@ -0,0 +1,32 @@
+## 93: Linq to EF Join throws ArgumentOutOfRangeException
```

Comment:
{
  "Id": "93",
  "Title": "Linq to EF Join throws ArgumentOutOfRangeException",
  "Details": "Linq queries or enumerables that call 'Join' on Entity Framework entities can cause an `ArgumentOutOfRangeException` in .NET Framework 4.5",
  "Suggestion": "This issue was fixed in a servicing update. Please update the .NET Framework 4.5, or upgrade to .NET Framework 4.5.1 or later, to fix this issue.",
  "Link": "http://connect.microsoft.com/VisualStudio/feedback/details/763986/linq-to-ef-join-throws-index-was-out-of-range-after-upgrade-from-vs2010-to-vs2012",
  "Markdown": null,
  "ApplicableApis": [
    "M:System.Linq.Enumerable.Join`4(System.Collections.Generic.IEnumerable{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3})",
    "M:System.Linq.Enumerable.Join`4(System.Collections.Generic.IEnumerable{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3},System.Collections.Generic.IEqualityComparer{`2})",
    "M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3})",
    "M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3},System.Collections.Generic.IEqualityComparer{`2})",
    "M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Linq.ParallelQuery{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3})",
    "M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Linq.ParallelQuery{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3},System.Collections.Generic.IEqualityComparer{`2})",
    "M:System.Linq.Queryable.Join`4(System.Linq.IQueryable{`0},System.Collections.Generic.IEnumerable{`1},System.Linq.Expressions.Expression{System.Func{`0,`2}},System.Linq.Expressions.Expression{System.Func{`1,`2}},System.Linq.Expressions.Expression{System.Func{`0,`1,`3}})",
    "M:System.Linq.Queryable.Join`4(System.Linq.IQueryable{`0},System.Collections.Generic.IEnumerable{`1},System.Linq.Expressions.Expression{System.Func{`0,`2}},System.Linq.Expressions.Expression{System.Func{`1,`2}},System.Linq.Expressions.Expression{System.Func{`0,`1,`3}},System.Collections.Generic.IEqualityComparer{`2})"
  ],
  "Related": null,
  "VersionBroken": "4.5",
  "VersionFixed": "4.5",
  "IsBuildTime": false,
  "IsQuirked": false,
  "IsSourceAnalyzerAvailable": false,
  "BugLink": null,
  "Notes": null,
  "IsRetargeting": false,
  "ImpactScope": 2
}

**Parsed APIs:**
- [ ] M:System.Linq.Enumerable.Join`4(System.Collections.Generic.IEnumerable{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3})
- [ ] M:System.Linq.Enumerable.Join`4(System.Collections.Generic.IEnumerable{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3},System.Collections.Generic.IEqualityComparer{`2})
- [ ] M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3})
- [ ] M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Collections.Generic.IEnumerable{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3},System.Collections.Generic.IEqualityComparer{`2})
- [ ] M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Linq.ParallelQuery{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3})
- [ ] M:System.Linq.ParallelEnumerable.Join`4(System.Linq.ParallelQuery{`0},System.Linq.ParallelQuery{`1},System.Func{`0,`2},System.Func{`1,`2},System.Func{`0,`1,`3},System.Collections.Generic.IEqualityComparer{`2})
- [ ] M:System.Linq.Queryable.Join`4(System.Linq.IQueryable{`0},System.Collections.Generic.IEnumerable{`1},System.Linq.Expressions.Expression{System.Func{`0,`2}},System.Linq.Expressions.Expression{System.Func{`1,`2}},System.Linq.Expressions.Expression{System.Func{`0,`1,`3}})
- [ ] M:System.Linq.Queryable.Join`4(System.Linq.IQueryable{`0},System.Collections.Generic.IEnumerable{`1},System.Linq.Expressions.Expression{System.Func{`0,`2}},System.Linq.Expressions.Expression{System.Func{`1,`2}},System.Linq.Expressions.Expression{System.Func{`0,`1,`3}},System.Collections.Generic.IEqualityComparer{`2})